### ISSUE TITLE
Consider absl::string_view

### ISSUE THREAD

**Issue by llchan** [artpaul/clickhouse-cpp#67]
I'd like to use clickhouse-cpp, but can't use C++17 at the moment.  I don't think C++17 adoption is very high yet overall, so it may be worth supporting at least C++14 (if not C++11).  Would you be open to using a polyfill like `absl::string_view`?  It uses `std::string_view` if available, and if not provides a pretty well-tested implementation.

**Comment by artpaul** [comment]
Hi,
do you have problem with this commit?
https://github.com/artpaul/clickhouse-cpp/commit/485ffeb89e8321af7572e2cba9b1ff79a8f6ff14

**Comment by llchan** [comment]
Yeah, more or less. I'd like to use clickhouse-cpp in a project that cannot be upgraded to C++17 quite yet.  Ideally we would use a string_view polyfill if you're open to it?  `absl::string_view` is probably the most mature one that I've seen, and you are worried about the third party dep doing something unexpected, you can gate it using `__cplusplus` so newer versions are guaranteed to use the stdlib one if C++17.

**Comment by Dlougach** [comment]
@llchan There are some problems with manual gating approach, especially if you want to create a library that can be linked (i.e. ABI-compatible) with all of C++11/14/17. absl solves this problem by having a special config header where you can force it to always use its own implementation regardless of the C++ standard.
